daemon/list.go

```diff
@@ -166,6 +176,10 @@ func (daemon *Daemon) reduceContainers(config *types.ContainerListOptions, reduc
 	// end up locking and querying many more containers than intended
 	containerList := daemon.filterByNameIDMatches(ctx)
 
+	// We have to do the sort here as types.Container's Created can only
+	// give us Creation time  in second
+	sort.Sort(sort.Reverse(byContainerCreated(containerList)))
```

Comment:
Given that the containers *only* get out of order if the list is filtered by Name (or ID) in `filterByNameIDMatches`, I think this should be moved to line 162 (before `return cntrs`)